bin/deferred_work_manager

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env ruby
+
+require "tom_queue"
+
+module TomQueue
+  class Daemon
+    include LoggingHelper
+
+    def run(argv = ARGV)
+      return unless load_rails!(".")
+
+      deferred_manager = DeferredWorkManager.new
+      deferred_manager.start
+    end
+
+    # Load Rails environment
+    #
+    # Returns true/false indicate whether Rails is loaded or not
+    def load_rails!(path)
+      rails_path = File.expand_path(File.join(path, 'config/environment.rb'))
+      if File.exists?(rails_path)
+        info "Loading rails from `#{rails_path}`"
+
+        ENV['RACK_ENV'] ||= ENV['RAILS_ENV'] ||= 'development'
+        require rails_path
+        ::Rails.application.eager_load!
+        true
+      else
+        false
+      end
```

Comment:
Yeah, all this feels like it has been lifted from somewhere unnecessarily - we don't have the use-cases or explicit testing to support any of this behaviour - so it's just complicating it with no benefit right now. Can we just fix the signal handling?